vite.config.js

```diff
@@ -1,9 +1,72 @@
-import { defineConfig } from "vite";
+import {defineConfig} from "vite";
 import react from "@vitejs/plugin-react";
 import svgr from "vite-plugin-svgr";
-import { visualizer } from "rollup-plugin-visualizer";
+import {resolve} from 'path';
+import fs from "node:fs/promises";
+import {devServerHtmlScript} from './scripts/dev-server-html-script.js';
 
-// https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [react(), svgr(), visualizer()],
+export default defineConfig(({command, mode, isSsrBuild, isPreview}) => {
+    const defaultConfig = {
+        plugins: [react(), svgr(), {
+            name: 'index-html-build-replacement', apply: 'serve', //   docs:
+            //   https://vitejs.dev/guide/api-plugin.html#conditional-application
+            async transformIndexHtml(html) {
+                const tags = [];
+                if (command === 'serve') {
```

Comment:
@alonisser This is the most elegant way I could find of injecting the script only on dev environment.